grammar.js

```diff
@@ -476,26 +480,21 @@ module.exports = grammar({
       field('right', $._expression)
     ),
 
-    inc_statement: $ => seq(
+    incdec_statement: $ => seq(
       $._expression,
-      '++'
-    ),
-
-    dec_statement: $ => seq(
-      $._expression,
-      '--'
+      field('operator', choice('++', '--'))
     ),
 
+    // The LHS of an assignment is restricted to:
+    //  lhs = id | '*' expr | expr '[' expr ']' | expr '.' id | '(' lhs ')'
```

Comment:
This is a pretty minor and subjective thing, but personally, I don't feel like we need to add comments explaining semantic restrictions like this that aren't modeled in the grammar. I think comments mostly are useful when explaining some surprising bit of complexity.